src/mumble/ServerHandler.cpp

```diff
@@ -699,10 +699,16 @@ void ServerHandler::disconnect() {
 
 void ServerHandler::serverConnectionClosed(QAbstractSocket::SocketError err, const QString &reason) {
 	Connection *c = cConnection.get();
-	if (!c)
+	if (!c) {
+		exit(0);
 		return;
-	if (c->bDisconnectedEmitted)
+	}
+
+	if (c->bDisconnectedEmitted) {
+		exit(0);
```

Comment:
This seems like it could terminate the cleanup process happening below if this callback is called twice. Not sure if that's a good idea 🤔